### ISSUE TITLE
Change in Perl 5 blead triggers failures in two Mojolicious tests

### ISSUE THREAD

**Issue by jkeenan** [mojolicious/mojo#2194]
As demonstrated by recent CPANtesters reports (*e.g.,* http://www.cpantesters.org/cpan/report/1888d746-591f-11ef-a424-e9834f22f64f), a change in the "blead" development branch of the Perl core distribution has triggered failures in tests in the Mojolicious distribution.  For example:
```
PERL_DL_NONLAZY=1 "/home/cpan/bin/perl/bin/perl5.41.3" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t t/*/*.t
t/mojo/asset.t ............................. ok
Can't locate Mojo/BaseTestTest.pm in @INC (you may need to install the Mojo::BaseTestTest module) (@INC entries checked: /home/cpan/.cpan/build/Mojolicious-9.37-0/t/mojo/lib /home/cpan/.cpan/build/Mojolicious-9.37-0/blib/lib /home/cpan/.cpan/build/Mojolicious-9.37-0/blib/arch /home/cpan/bin/perl/lib/site_perl/5.41.3/amd64-freebsd /home/cpan/bin/perl/lib/site_perl/5.41.3 /home/cpan/bin/perl/lib/5.41.3/amd64-freebsd /home/cpan/bin/perl/lib/5.41.3 .) at /home/cpan/.cpan/build/Mojolicious-9.37-0/blib/lib/Mojo/Base.pm line 132.
BEGIN failed--compilation aborted at t/mojo/base.t line 22.
t/mojo/base.t .............................. 
Dubious, test returned 2 (wstat 512, 0x200)
No subtests run 
...
t/mojo/json_xs.t ........................... ok

    #   Failed test 'loaded successfully'
    #   at t/mojo/loader.t line 79.

    #   Failed test 'loaded successfully'
    #   at t/mojo/loader.t line 80.
    # Looks like you failed 2 tests of 4.

#   Failed test 'Load'
#   at t/mojo/loader.t line 84.
# Looks like you failed 1 test of 15.
t/mojo/loader.t ............................ 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/15 subtests 
t/mojo/log.t ............................... ok
```
The change in Perl 5 blead in question was the final removal of the long deprecated use of an apostrophe (`'`) as a package separator.  (See: https://github.com/Perl/perl5/pull/22303.)  The implication (my interpretation) is that for Perl versions starting with development release 5.41.3, scheduled for August 20 2024, tests should no longer expect package names using apostrophes to pass, so such tests should be skipped.

**Comment by kraih** [comment]
#2193

**Comment by Grinnz** [comment]
Resolved by #2193
